File: basic/java/1001. 害死人不偿命的(3n+1)猜想 (15).java
import java.util.Scanner;

public class Main {
	public static void main(String args[]) {
		int n,res=0;
		Scanner in=new Scanner(System.in);
		n=in.nextInt();
		in.close();
		while(n!=1) {
			if(n%2!=0)n=3*n+1;
			n/=2;
			res++;
		}
		System.out.println(res);
	}
}

File: basic/1001. 害死人不偿命的(3n+1)猜想 (15).cpp
#include<iostream>
using namespace std;
int main(){
	int n,res=0;
	cin>>n;
	while(n!=1){
		if(n%2)n=n*3+1;
		n/=2;
		++res;
	}
	cout<<res;
	return 0;
}
